cli/helpers/maven/base_restore_maven_dependencies.py

```diff
@@ -0,0 +1,77 @@
+from abc import ABC, abstractmethod
+
+import click
+from typing import List, Optional, Dict
+
+from cli.models import Document
+from cli.utils.path_utils import join_paths, get_file_dir
+from cli.utils.shell_executor import shell
+from cyclient import logger
+
+
+class BaseRestoreMavenDependencies(ABC):
+    context: click.Context
+    documents_to_add: List[Document]
+    is_git_diff: bool
+    command_timeout: int
+
+    def __init__(self, context: click.Context, documents_to_add: List[Document], is_git_diff: bool,
+                 command_timeout: int):
+        self.context = context
+        self.documents_to_add = documents_to_add
+        self.is_git_diff = is_git_diff
+        self.command_timeout = command_timeout
+
+    def restore(self, document: Document):
+        if self.is_project(document):
+            restore_dependencies_document = self.try_restore_dependencies(document)
+            if restore_dependencies_document is None:
+                logger.warning('Error occurred while trying to generate dependencies tree. %s',
+                               {'filename': document.path})
+                return
+            manifest_file_path = self.get_manifest_file_path(document)
+            if restore_dependencies_document.content is None:
+                logger.warning('Error occurred while trying to generate dependencies tree. %s',
+                               {'filename': document.path})
+                restore_dependencies_document.content = ''
+                logger.debug(
+                    f"Failed to generate dependencies tree on path: {manifest_file_path}")
+            else:
+                logger.debug(f"Succeeded to generate dependencies tree on path: {manifest_file_path}")
+            self.documents_to_add.append(restore_dependencies_document)
+
+    def get_manifest_file_path(self, document: Document) -> str:
+        return join_paths(self.context.params.get('path'), document.path) if self.context.obj.get(
+            'monitor') else document.path
+
+    def build_dep_tree_path(self, path: str, generated_file_name: str) -> str:
```

Comment:
@morsa4406 this method just calling join path. in my opinion you have 2 options:
1. remove it at all and instead of calling it call directly to join_paths
2. put it inside the class and call it _build_dep_tree (In python it indicates private method)